src/components/Search/CompareWithBase.tsx

```diff
@@ -99,6 +92,8 @@ function CompareWithBase({
     useState<Changeset[]>(newRevs);
   const [baseRepository, setBaseRepository] = useState(baseRepo);
   const [newRepository, setNewRepository] = useState(newRepo);
+  const [formIsDisplayed, setFormIsDisplayed] = useState(!hasEditButton);
+  const [hasCancelButton, setHasCancelButton] = useState(false);
```

Comment:
optional:
I think this state about cancelButton isn't useful and can be simply computed from these 2 other states:
```js
const hasCancelButton = formIsDisplayed && hasEditButton
```